cicd/_common_deploy_logic.sh

```diff
@@ -28,45 +32,56 @@ trap_proxy teardown EXIT ERR SIGINT SIGTERM
 
 set -e
 
-: ${COMPONENTS:=""}
-: ${COMPONENTS_W_RESOURCES:=""}
-: ${DEPLOY_TIMEOUT:="900"}
-: ${REF_ENV:="insights-production"}
-: ${RELEASE_NAMESPACE:="true"}
-: ${ALWAYS_COLLECT_LOGS:="false"}
+: "${COMPONENTS:=}"
+: "${COMPONENTS_W_RESOURCES:=}"
+: "${DEPLOY_TIMEOUT:=900}"
+: "${REF_ENV:=insights-production}"
+: "${RELEASE_NAMESPACE:=true}"
+: "${ALWAYS_COLLECT_LOGS:=false}"
 
 K8S_ARTIFACTS_DIR="$ARTIFACTS_DIR/k8s_artifacts"
 TEARDOWN_RAN=0
 
 function get_pod_logs() {
-    local ns=$1
-    LOGS_DIR="$K8S_ARTIFACTS_DIR/$ns/logs"
-    mkdir -p $LOGS_DIR
+
+    local NAMESPACE="$1"
+    local LOGS_DIR="${K8S_ARTIFACTS_DIR}/${NAMESPACE}/logs"
+    local F_IFS="$IFS"
+
+    mkdir -p "$LOGS_DIR"
+
     # get array of pod_name:container1,container2,..,containerN for all containers in all pods
     echo "Collecting container logs..."
-    PODS_CONTAINERS=($(oc_wrapper get pods --ignore-not-found=true -n $ns -o "jsonpath={range .items[*]}{' '}{.metadata.name}{':'}{range .spec['containers', 'initContainers'][*]}{.name}{','}"))
```

Comment:
IMO this was more easily understandable compared to what's now on line 56 w/ the IFS and 'read' stuff. What do you think?